lib/fluent/plugin/in_azureactivitylog.rb

```diff
@@ -0,0 +1,129 @@
+require 'fluent/input'
+require 'azure_mgmt_monitor'
+require 'uri'
+
+class Fluent::AzureActivityLogInput < Fluent::Input
+  Fluent::Plugin.register_input("azureactivitylog", self)
+
+  # To support log_level option implemented by Fluentd v0.10.43
+  unless method_defined?(:log)
+    define_method("log") { $log }
+  end
+
+  # Define `router` method of v0.12 to support v0.10 or earlier
+  unless method_defined?(:router)
+    define_method("router") { Fluent::Engine }
+  end
+
+  config_param :tag,                :string
+  config_param :tenant_id,          :string, :default => nil
+  config_param :subscription_id,    :string, :default => nil
+  config_param :client_id,          :string, :default => nil
+  config_param :client_secret,      :string, :default => nil, :secret => true
+
+  config_param :select_filter,      :string, :default => nil
+  config_param :event_channels,     :string, :default => "Admin, Operation"
+  config_param :interval,           :integer, :default => 300
+
+  def initialize
+    super
+  end
+
+  def configure(conf)
+    super
+
+    provider = MsRestAzure::ApplicationTokenProvider.new(@tenant_id, @client_id, @client_secret)
+    credentials = MsRest::TokenCredentials.new(provider)
+    @client = Azure::ARM::Monitor::MonitorManagementClient.new(credentials);
+    @client.subscription_id = @subscription_id
+  end
+
+  def start
+    super
+    @watcher = Thread.new(&method(:watch))
+  end
+
+  def shutdown
+    super
+    @watcher.terminate
+    @watcher.join
+  end
+
+  private
+
+  def watch
+    while true
+        log.debug "azure activitylog: watch thread starting"
```

Comment:
check general debug setting

Reply:
I checked it.
Debug messages will be shown only when -vv flag is added to the fluent command.